deflate.h

```diff
@@ -327,14 +327,15 @@ static inline void put_uint32(deflate_state *s, uint32_t dw) {
  * IN assertion: there is enough room in pending_buf.
  */
 static inline void put_uint32_msb(deflate_state *s, uint32_t dw) {
-#if defined(UNALIGNED_OK)
-    *(uint32_t *)(&s->pending_buf[s->pending]) = ZSWAP32(dw);
+#if BYTE_ORDER == LITTLE_ENDIAN
+    uint32_t msb = ZSWAP32(dw);
+    memcpy(&s->pending_buf[s->pending], &msb, 4);
+    s->pending += 4;
+#elif BYTE_ORDER == BIG_ENDIAN
+    memcpy(&s->pending_buf[s->pending], &dw, 4);
     s->pending += 4;
 #else
-    put_byte(s, ((dw >> 24) & 0xff));
-    put_byte(s, ((dw >> 16) & 0xff));
-    put_byte(s, ((dw >> 8) & 0xff));
-    put_byte(s, (dw & 0xff));
+#  error No endian defined
```

Comment:
Should be `No endianess defined`.

Reply:
I copied the text from functable.c

Reply:
I haven't reviewed all files... There is three long-time contributors and only two need to review a pull request before it gets merged.